### PR TITLE
Add UUID typing annotation instead of strings everywhere

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
Add uuid type annotation to all id fields that are actually always uuids.

### Issue link
Closes https://github.com/minvws/nl-kat-coordination/issues/1451

### Proof
See tests.

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/tests/conftest.py [hunk 1]
```diff
@@ -61,22 +62,22 @@ def pop_item(self, queue: str) -> Optional[QueuePrioritizedItem]:
         except IndexError:
             raise self.raise_on_empty_queue
 
-    def patch_task(self, task_id: str, status: TaskStatus) -> None:
+    def patch_task(self, task_id: UUID, status: TaskStatus) -> None:
         with self.log_path.open("a") as f:
-            f.write(f"{task_id},{status.value}\n")
+            f.write(f"{str(task_id)},{status.value}\n")
```
Comment (ammar92) [inline boefjes/tests/conftest.py:67]: Aren't objects other than strings implicitly cast to a `str` within f-strings?

**Review by ammar92 — APPROVED** [review]
Looks good to me. Just a remark/ question about implicit casting in f-strings

File: boefjes/tests/conftest.py (hunk 1 above)
Comment (Donnype) [reply boefjes/tests/conftest.py:67]: Yes this is probably redundant

Comment (Donnype) [reply boefjes/tests/conftest.py]: Fixed

**Review by ammar92 — APPROVED** [review]

**Comment by ammar92** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [ ] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
- Viewing tasks and scheduling works

**Review by dekkers — APPROVED** [review]

**Pull request merged**